languages/js/src/Query.ts

```diff
@@ -174,6 +175,10 @@ export class Query {
             // Error on attempt to call non-function.
             throw new InvalidCallError(receiver, attr);
           }
+        } else {
+          if (value === undefined && !(attr in receiver)) {
+            throw new InvalidAttributeError(receiver, attr)
+          }
         }
       }
     } catch (e) {
```

Comment:
Should we also check `e instanceof InvalidAttributeError` on the line below this one?